src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeReader.cs

```diff
@@ -158,13 +152,15 @@ public virtual Task CopyToAsync(Stream destination, CancellationToken cancellati
                 return Task.FromCanceled(cancellationToken);
             }
 
-            return CopyToAsyncCore(
-                destination,
-                (destination, memory, cancellationToken) => destination.WriteAsync(memory, cancellationToken),
-                cancellationToken);
+            return CopyToAsyncCore(destination, async (destination, memory, cancellationToken) =>
+            {
+                await destination.WriteAsync(memory, cancellationToken).ConfigureAwait(false);
```

Comment:
You can avoid the state machine if this finishes synchronously.


```suggestion
                var task = destination.WriteAsync(memory, cancellationToken).ConfigureAwait(false);
                
                static async Task Awaited(Task writeTask)
                {
                    await writeTask;
                    return new FlushResult(isCanceled: false, isCompleted: false);
                }
                
                if (task.IsCompletedSuccessfully)
                {
                    task.GetAwaiter().GetResult();
                    return new FlushResult(isCanceled: false, isCompleted: false);
                }
                return Awaited(task);
```